pkg/server/wire.go

```diff
@@ -325,7 +328,6 @@ var wireBasicSet = wire.NewSet(
 	grpcserver.ProvideHealthService,
 	grpcserver.ProvideReflectionService,
 	interceptors.ProvideAuthenticator,
-	setting.NewCfgFromArgs,
```

Comment:
This "dependency" only exists so wire can handle calling that method on the setting.CommandLineArgs, so I refactored the various Initialize commands to take a `setting.Cfg` and made sure we were calling `setting.NewCfgFromArgs` explicitly where needed. 

This does not change the setting.Provider (oss & enterprise both); that is a more idiomatic dependency.

It's an unexpected, free baby step towards refactoring setting.Cfg!!